src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -182,6 +182,38 @@ object Scanners {
      */
     var revComments: List[Comment] = Nil
 
+    /** The currently closest docstring, replaced every time a new docstring is
+     *  encountered
+     */
+    var closestDocString: List[List[Comment]] = List(List())
```

Comment:
That was my original thought, but I wasn't sure how much of the standard library stuff we could use - but I'll change it to a stack on your recommendation.